tests/test_deepcopy.py

```diff
@@ -0,0 +1,35 @@
+from copy import deepcopy
+import pytest
+
+import tensorflow as tf
+import tensorflow_probability as tfp
+from gpflow.utilities import deepcopy_components
+
+
+
+class A(tf.Module):
+    def __init__(self):
+        self.var = tf.Variable(tf.random.normal((1,)))
+        self.bijector = tfp.bijectors.Softplus()
+
+    def __call__(self, x):
+        return self.bijector(x)
+
+
+class B(tf.Module):
+    def __init__(self):
+        self.var = tf.Variable(tf.random.normal((1,)))
+        self.a = A()
+
+    def __call__(self, x):
+        return self.a(x)
+
+
+@pytest.mark.parametrize('module', [A(), B()])
+def test_deepcopy_component_works(module):
+    input = 1.
+    _ = module(input)
+    with pytest.raises(TypeError):
+        deepcopy(module)
```

Comment:
why should this raise a `TypeError`? could you add a comment explaining why that would be expected to not work?

Reply:
I added an explanation on the docstring of the test :)

Reply:
We don't raise error anymore. How does this test pass? :)

Reply:
it's calling python stdlib `deepcopy` not the custom `deepcopy_components`

Reply:
TypeError corresponds to the error raised by standard `deepcopy` when copying a bijector with cache. I explain it in the docstring.